src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs

```diff
@@ -147,11 +313,107 @@ public string GetLinkByAddress<TAddress>(TAddress address, HttpContext httpConte
             throw new InvalidOperationException("Could not find a matching endpoint to generate a link.");
         }
 
+        /// <summary>
+        /// Tries generating a URL with an absolute path.
+        /// </summary>
+        /// <param name="address">The information used to look up endpoints for generating a URL.</param>
+        /// <param name="httpContext">The <see cref="HttpContext"/> associated with current request.</param>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="options">The <see cref="LinkOptions"/>.</param>
+        /// <param name="link">The generated absolute URL.</param>
+        /// <returns><c>true</c> if a URL was generated successfully, <c>false</c> otherwise.</returns>
         public abstract bool TryGetLinkByAddress<TAddress>(
             TAddress address,
             HttpContext httpContext,
             object values,
             LinkOptions options,
             out string link);
+
+        /// <summary>
+        /// Creates a template object to generate a URL.
```

Comment:
nit: Summary says create, method is called Get. I view them as different things.

Change this to "Gets a `LinkGenerationTemplate` for an endpoint matching the specified route values."?